src/amount.rs

```diff
@@ -208,6 +208,14 @@ mod test {
         assert_eq!(1, microtez.try_into_minor_units().unwrap());
     }
 
+    fn balance_parsing_helper<T>(amount: &str)
```

Comment:
nit: I'd prefer a more descriptive name, like `try_parse_invalid_balance` or `incorrect_amount_cannot_parse`. This name doesn't quite explain what help the function is providing, but I think it's important to flag that the helper is for _invalid_ amount strings and will mess up the tests if you put a reasonable value in there.